src/host/ledger.h

```diff
@@ -246,6 +246,14 @@ namespace asynchost
           "Failed to read positions offset from ledger file {}", file_path));
       }
 
+      if (committed && table_offset == 0)
+      {
+        throw std::logic_error(fmt::format(
+          "Invalid table offset at start of committed ledger file {}. Is file "
+          "corrupt?",
```

Comment:
```suggestion
          "Committed ledger file {} cannot be read: invalid table offset (0)",
```

I think it's a definite that the ledger file is corrupt and unusable in this situation.